Add Workflow Examples and Improve Section/Reference Titles

SteVwonder opened the pull request:
Adds a subsection to `index.rst` that references the new flux-workflow-examples docs.
Improves the section and reference titles for flux-core and flux-rfcs.

Closes #75, #76

Depends on https://github.com/flux-framework/flux-core/pull/3365

cmoussa1 reviewed (approved):
LGTM! Built this on my local machine and the workflow examples section looks great. Thanks @SteVwonder! Just one unrelated note about our RTD site:

Definitely doesn't have to go in this PR, but I noticed on the side bar of the home page that we have links to the first few sections of the home page (i.e links to "Quick Start", "Flux Administrator's Guide", ... up to "FAQs"), but not the sections that follow ("Contributor Relevant RFC's", "All RFC's", ... up to "Indices and Tables"). It might be nice to at one point include these other sections near the bottom of the home page for quick and easy access. 🙂

SteVwonder commented:
> I noticed on the side bar of the home page that we have links to the first few sections of the home page (i.e links to "Quick Start", "Flux Administrator's Guide", ... up to "FAQs"), but not the sections that follow ("Contributor Relevant RFC's", "All RFC's", ... up to "Indices and Tables"). 

That's a great suggestion.  I went ahead and added links to the various docs sub-projects (core, rfcs, examples) in the sidebar.  Let me know what you think.

I also fixed up the reference to the man pages in flux-core now that that PR has been merged.

cmoussa1 commented:
Awesome! This looks great. Thanks @SteVwonder. Feel free to set MWP when you are ready.
